clang/include/clang/Parse/Parser.h

```diff
@@ -467,15 +467,18 @@ class Parser : public CodeCompletionHandler {
 
   /// Flags describing a context in which we're parsing a statement.
   enum class ParsedStmtContext {
+    /// This context permits declarations in language modes where declarations
+    /// are not statements.
+    AllowDeclarationsInC = 0x1,
```

Comment:
This feels a little mysterious, I only see that we check it but it is not obvious how it gets set.

Reply:
@shafik L481 below